adage/analyze/views.py

```diff
@@ -74,13 +75,36 @@ class SignatureViewSet(ReadOnlyModelViewSet):
 
 
 class EdgeViewSet(ReadOnlyModelViewSet):
-    """Gene-gene edge viewset. Supported parameter: `mlmodel`"""
+    """
+    Gene-gene edge viewset.
+    Supported parameter: `mlmodel`, `genes`.
+    """
 
     http_method_names = ['get']
-    queryset = Edge.objects.all()
     serializer_class = EdgeSerializer
     filterset_fields = ['mlmodel', ]
 
+    def get_queryset(self):
+        queryset = Edge.objects.all()
+        genes = self.request.query_params.get('genes', None)
+        if genes:
+            try:
+                gene_ids = {int(id) for id in genes.split(',')}
+            except ValueError:
+                raise ParseError(
+                    {'error': f'Invalid gene IDs: {genes}'}
+                )
+            qset = Q(gene1__in=gene_ids) | Q(gene2__in=gene_ids)
+            direct_edges = queryset.filter(qset).distinct()
```

Comment:
It's possible this filter command is silently doing nothing since a union has been applied prior.

https://code.djangoproject.com/ticket/30639
https://code.djangoproject.com/ticket/27995

In django 3 it should raise an error as per https://github.com/django/django/commit/1853383969a4c53bbeba998757c30410bd3df4bb

Reply:
I did some test and confirmed that this bug wouldn't affect my code. It would affect the following code:
```python
qs1 = queryset.filter(gene1__in=gene_ids)
qs2 = queryset.filter(gene2__in=gene_ids)
qset = qs1.union(qs2)
```
which calls `union` method explicitly.